File: camera.py
import cv2

# カメラデバイスの取得（デフォルトでID 0を使用）
cap = cv2.VideoCapture(0)

def main():
    # フレームの取得
    ret, frame = cap.read()
    
    # フレームが正しく取得できたか確認
    if not ret:
        return

    # フレームを表示
    cv2.imshow('frame', frame)

    # 'q' キーが押されたらループを終了
    if cv2.waitKey(1) & 0xFF == ord('q'):
        return

    # カメラデバイスを解放
    cap.release()
    cv2.destroyAllWindows()

if __name__ == "__main__":
    main()

File: 17_DebriCheck/Camera/Camera.cpp
#include "Camera.hpp"
#include <stdio.h>

using namespace cv;

Camera::Camera()
{
	cap.open(0);
	if (!cap.isOpened())
	{
		printf("カメラが開けません\n");
	}
}

Camera::~Camera()
{
	destroyAllWindows();
    cap.release();
}

void Camera::captureVideo()
{
	cap >> frame;
	if (frame.empty())
	{
		printf("フレームが空です\n");
		return;
	}
}

void Camera::showVideo()
{
	imshow("camera", frame);
	waitKey(1);
}

void Camera::getVideo(Mat& frm) const
{
	frm = frame;
}
